src/managers/GuildMemberManager.js

```diff
@@ -146,16 +147,28 @@ class GuildMemberManager extends BaseManager {
    * guild.members.prune({ days: 1, reason: 'too many people!' })
    *   .then(pruned => console.log(`I just pruned ${pruned} people!`))
    *   .catch(console.error);
+   * @example
+   * // Include members with a specified role
+   * guild.members.prune({ days: 7, roles: ['657259391652855808'] })
+   *    .then(pruned => console.log(`I just pruned ${pruned} people!`))
+   *    .catch(console.error);
    */
-  prune({ days = 7, dry = false, count = true, reason } = {}) {
+  prune({ days = 7, dry = false, count = true, roles = [], reason } = {}) {
     if (typeof days !== 'number') throw new TypeError('PRUNE_DAYS_TYPE');
+
+    const query = new URLSearchParams();
+    query.set('days', days);
+    query.set('compute_prune_count', count);
+
+    for (const role of roles) {
+      const resolvedRole = this.guild.roles.resolveID(role);
+      if (resolvedRole) query.append('include_roles', role);
```

Comment:
should some sort of error be thrown here if the role is invalid?

Reply:
impl here:  
https://github.com/Fyko/discord.js/blob/2c86024f4a9fed9820ca4cade1e986693ffcb40a/src/managers/GuildMemberManager.js#L165-L167

Reply:
The if check is redundant now